.github/workflows/pr.yaml

```diff
@@ -23,5 +24,62 @@ jobs:
     with:
       enable_check_generated_files: false
       enable_check_pr_job_dependencies: false
-  build-and-test-wheels:
-    uses: ./.github/workflows/build_and_test_wheels.yaml
+  compute-matrix-build:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix-build.outputs.MATRIX }}
+    steps:
+      - name: Compute Build Matrix
+        id: compute-matrix-build
+        run: |
+          set -eo pipefail
+
+          export MATRIX="
+          - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.9', LINUX_VER: 'centos7' }
+          - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.10', LINUX_VER: 'centos7' }
+          - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.9', LINUX_VER: 'rockylinux8' }
+          - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.10', LINUX_VER: 'rockylinux8' }
+          "
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '.' \
+          )" | tee --append "${GITHUB_OUTPUT}"
+  compute-matrix-test:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix-test.outputs.MATRIX }}
+    steps:
+      - name: Compute Build Matrix
+        id: compute-matrix-test
+        run: |
+          set -eo pipefail
+
+          export MATRIX="
+          - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.9', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+          - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.10', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+          - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.9', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+          - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.10', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+          "
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '.' \
+          )" | tee --append "${GITHUB_OUTPUT}"
+  build-wheels:
+    needs:
+      - compute-matrix-build
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.02
+    with:
+      build_type: pull-request
+      script: "ci/build_wheel.sh"
+      matrix_filter: ${{ needs.compute-matrix-build.outputs.MATRIX }}
+  test-wheels:
+    needs:
+      - build-wheels
+      - compute-matrix-test
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.02
```

Comment:
TODO: reuse `wheels-build` to test on CPU machines

Reply:
This resulted in the errors below, so we reverted to testing on GPU machines:

- https://github.com/rapidsai/pynvjitlink/actions/runs/7049358912/job/19187956792#step:9:110

We can revisit this in the future.

Marking as resolved for now.